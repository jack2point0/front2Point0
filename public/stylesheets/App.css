* {
    font-family:"Open Sans","Helvetica Neue",Helvetica,Arial,sans-serif;
}

html {
  transition: 0.5s;
  background-color: #292f36;
}

a {
    text-decoration: none;
}

a:hover {
    color: #CF5300;
    text-decoration: none;
}

.form-control {
    border-radius: 5px;
}

.b {
  background-color: #292f36;
  min-height: 100vh;
  position: relative;
  transition: 0.5s
}

input,button {
  font-family: sans-serif;
  color: #292f36;
}

.table {
  display: flex;
  flex-direction: column;
  /* border: 2px solid black; */
  align-items: center;
  color: #EFE9E7;
}

.bodyContainer {
display: flex;
flex-direction: row;
justify-content: center;
/* border: 0.5px solid blue; */
max-width: 1220px;
flex-wrap: wrap;
}

.box {
  margin-top: 5px;
  margin-bottom: 5px;
  width: 255px;
}

input {
  width: 150px;
}
.testing {
  width: 80vw;
}

.tab-pane.fade {
  display: flex;
  flex-wrap: wrap;
}

.tabsContainer {
display: flex;
flex-direction: column;
max-width: 1300px;
flex-wrap: wrap;
justify-content: space-around;
}

.buttonDiv {
  display: flex;
  flex-direction: row;
  justify-content: center;
  justify-content: space-around;
  margin: 50px;
}

#notesField {
  margin-bottom: 40px;
}

.errors {
  color: #CF5300;
  font-weight: bold;
  padding-left: 120px;
}

.errors::first-letter {
  text-transform: capitalize;
}

.greeting {
  padding-left: 40px;
  margin-bottom: 20px;
  font-weight: 500;
  color: #CF5300;
  border-bottom: 1px solid #EFE9E7;
  margin-left: 50px;
  margin-right: 50px;
}


@media(max-width: 678px) {
    .tabsContainer {
        max-width: 405px;
    }
}
