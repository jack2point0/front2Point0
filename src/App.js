import React, { Component } from 'react';
import Header from './Components/Header.js'
import Footer from './Components/Footer.js'
import Dashboard from './Pages/Dashboard.js'
import Home from './Pages/Home.js'
import Instructions from './Pages/Instructions.js'
import Registration from './Pages/Registration.js'
import Sign_In from './Pages/Sign_in.js'
import Task_Index from './Pages/Task_index.js'
import { Redirect, BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import AuthService from './services'
import { getMyTasks } from './API';
import Appointments from './Pages/Appointments'
import AptForm from './Pages/aptForm'
import EditMyTaskCard from './Components/editMyTaskCard'
import Profile from './Pages/Profile'


class App extends Component {
    constructor(props) {
      super(props)

      this.auth = new AuthService()
      this.state = {
        marginRight: 0,
        authenticated: this.auth.loggedIn(),
        hasToken: false,
        myTasks: [],
        userID: ''
      }
    }

  render() {
    return (
      <div style={{marginRight: this.state.marginRight}} className="b">
        <Header openSideMenu={this.openSideMenu}/>
        <div className="b">
          <Router>
            {
              (this.auth.loggedIn()) ?
              <Switch> //protected paths
                <Redirect from="/register" to="/" />
                <Redirect from="/login" to="/" />
                <Route exact path='/user/my_tasks/:id/edit' render={(routeProps) => <EditMyTaskCard refresh={this.refresh} {...routeProps}/>} />
                <Route exact path='/instructions' component={Instructions} />
                <Route exact path='/profile' render={(routeProps) => <Profile refresh={this.refresh} {...routeProps}/>} />
                <Route exact path='/tasks' render={(props) => <Task_Index userID={this.state.userID} />} />
                <Route exact path='/appointments' component={Appointments} />
                <Route exact path='/appointments/new' component={AptForm} />
                <Route path='/' render={(props) => <Dashboard myTasks={this.state.myTasks} />} />
              </Switch>

              :
              <Switch> //public paths
                <Route exact path='/login' render={(props) => <Sign_In refresh={this.refresh}/>} />
                <Route exact path='/register' render={(props) => <Registration refresh={this.refresh}/> }/>
                <Route path='/' component={Home} />
              </Switch>
            }
          </Router>
        </div>
          <footer >
            <Footer />
          </footer>
      </div>
    );
  }



  componentDidMount = () => {
    let thisUserID = this.auth.getUserId()
    if (thisUserID !== null && thisUserID !== undefined && thisUserID.length > 0) {
      getMyTasks(thisUserID)
      .then((APImyTasks)=> {
      // let { myTasks } = this.state
        this.setState({
          myTasks: APImyTasks,
          userID: thisUserID
        })
      })
    }
  }

  openSideMenu = () => {
    console.log("made it in");
    let { marginRight } = this.state
    marginRight = marginRight === 0 ? 200 : 0
    this.setState({
      marginRight: marginRight
    })
  }

  refresh = () => {
    this.setState ({
      authenticated: this.auth.loggedIn()
    })
  }
}

export default App;
