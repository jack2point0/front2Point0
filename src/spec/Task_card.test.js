import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import Task_Card from '../Components/Task_card';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'

Enzyme.configure({ adapter: new Adapter() });

const taskAtts = { title: 'this is a test', category: 'house', sub_category: 'cleaning' }

test('renders the Task_card correctly', () => {
    const json = renderer
        .create(<Task_Card info={taskAtts} />)
        .toJSON();
    expect(json).toMatchSnapshot();
})

// NOTE: fake testing below


// <div className="theContainer">
//     { this.props.info.title ? `<h2 className="title">${this.props.info.title }</h2>` : ' '}
// </div>

// test('the title should display a h2 element', () => {
//     const wrapper = shallow(<Task_Card info={taskAtts} />);
//     const containerShallow = wrapper.find('.theContainer')
// })
