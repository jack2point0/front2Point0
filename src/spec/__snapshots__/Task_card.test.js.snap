// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Task_card correctly 1`] = `
<div
  className="tile"
>
  <div>
    <div
      className="bodyBox box"
    >
      <div
        className="catInfoContainer"
      >
        <i />
        cleaning
      </div>
      <hr />
    </div>
    <div
      className="bodyBox box tileTitle"
    >
      this is a test
    </div>
    <div
      className="bodyBox box tileDescription"
    >
      <h5 />
    </div>
    <div
      className="tileButtonContainer"
    >
      <div
        className="tileAddBtn"
        onClick={[Function]}
      >
        Add
      </div>
    </div>
  </div>
</div>
`;
